ethereumj-core/src/main/java/org/ethereum/vm/DataWord.java

```diff
@@ -22,77 +22,122 @@
 import org.ethereum.db.ByteArrayWrapper;
 import org.ethereum.util.ByteUtil;
 import org.ethereum.util.FastByteComparisons;
-import org.spongycastle.util.Arrays;
 import org.spongycastle.util.encoders.Hex;
 
 import java.math.BigInteger;
 import java.nio.ByteBuffer;
+import java.util.Arrays;
 
 import static org.ethereum.util.ByteUtil.toHexString;
 
 /**
  * DataWord is the 32-byte array representation of a 256-bit number
  * Calculations can be done on this word with other DataWords
+ * DataWord is immutable. Use one of `of` factories for instance creation.
  *
  * @author Roman Mandeleil
  * @since 01.06.2014
  */
-public class DataWord implements Comparable<DataWord> {
+public final class DataWord implements Comparable<DataWord> {
```

Comment:
For some reason Github doesn't allow to comment on `public boolean isZero()` like. But this method lacks a shortcut check for whether `this == ZERO` or not